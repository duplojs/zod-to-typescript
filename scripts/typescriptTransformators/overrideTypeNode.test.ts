import { ZodToTypescript } from "@scripts/index";
import { factory } from "typescript";
import { z as zod } from "zod";

it("overrideTypeNode", () => {
	const zodSchema = zod.string().overrideTypeNode(
		factory.createTypeReferenceNode(factory.createIdentifier("RegExp")),
	);

	const result = ZodToTypescript.convert(zodSchema);

	expect(result).toMatchSnapshot();
});
