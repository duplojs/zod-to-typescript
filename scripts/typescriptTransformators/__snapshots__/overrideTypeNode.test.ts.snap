// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`overrideTypeNode 1`] = `"type Zod2ts_0_duplojs = RegExp;"`;
